db/compaction/tiered_compaction_test.cc

```diff
@@ -1111,12 +1112,18 @@ TEST_P(TieredCompactionTest, RangeBasedTieredStorageLevel) {
   options.num_levels = kNumLevels;
   options.statistics = CreateDBStatistics();
   options.max_subcompactions = 10;
+  // Prevent penultimate level from being smaller than
+  // max_bytes_for_level_base / 10 and deemed unnecessary.
+  // Otherwise L5 will always be eligible for compaction and
+  // WaitForCompact(true) below will be stuck waiting.
```

Comment:
There **will** be infinite rewrite of L5 since it always has a eligible compaction score but nothing is written to the last level during each compaction. This causes the `WaitForCompact(true)` in this unit test to wait forever.

Reply:
Hm that isn't ideal. We should see if there's a way to not pick a draining compaction when the output level is not necessarily lower.